tox.ini

```diff
@@ -0,0 +1,84 @@
+#
+# Setup and manage python virtual environments that allow users to work
+# globally with a set of coverage results each acquired individually.
+#
+# This tox.ini is explicitly setup in the root of the repository and saves the
+# aggerate coverage result to the root as .coverage.  This is done so that
+# Coveralls
+#   * finds the report,
+#   * shows correctly the filenames of python files inside (i.e., within the
+#     ibcdfo_pypkg directory) and outside (e.g., within subA) of the python
+#     package, and
+#   * allows users to view the coverage results for each python file inside and
+#     outside of the python package.
+# This extra care is needed as a result of developing subpackages such as subA
+# outside of the python package and including it into the package via symlink.
+#
+
+[tox]
+requires = tox>=4
+env_list =
+
+[flake8]
+max-line-length = 200
+ignore =
+        # E203 is not PEP 8 compliant. https://github.com/PyCQA/pycodestyle/issues/373
+        E203
+        # Not worrying about spaces around arithmetic operations
+        E226
+        # Linebreaks before/after unary operators are okay
+        W503, W504
+exclude =
+        # Don't need to check .git
+        .git,
+        ibcdfo_pypkg/build/,
+        minq5,
+        __init__.py,
+        *.tox/
+per-file-ignores =
+        # Allow lambda expressions
+        pounders/py/tests/*.py:E731
+        # These are old test
+        pounders/py/tests/old_unit_tests/test_pounders.py:F841,E501
+
+[testenv]
+basepython = py3
+skip_install = true
+setenv =
+    COVERAGE_FILE = .coverage
+
+[testenv:aggregate]
+description = Aggregate all given coverage results into single file
+deps = coverage
+commands = coverage combine --keep --data-file={env:COVERAGE_FILE} {posargs}
+
+[testenv:report]
+# Unless an individual coverage report was saved to COVERAGE_FILE, this should fail
+# if you don't run aggregate first
+description = Generate reports for aggregated coverage results
+deps = coverage
+depends = aggregate
+commands =
+    coverage report --data-file={env:COVERAGE_FILE}
+    coverage html --data-file={env:COVERAGE_FILE}
+
+[testenv:coveralls]
+# Unless an individual coverage report was saved to COVERAGE_FILE, this should fail
+# if you don't run aggregate first
+description = Post aggregated coverage results to coveralls via GitHub Action
+passenv = GITHUB_*, COVERALLS_*
+deps = coveralls
+depends = aggregate
+commands =
+    coveralls --service=github --basedir={toxinidir}
+    coveralls --finish
+
+[testenv:format]
+description = Clean-up code with black and reporting flake8 issues
+deps =
+    black
+    flake8
+skip_install = true
+commands =
+    black --config=.black {toxinidir}
+    flake8
```

Comment:
I don't know if I would include flake8 in `format`.  I understand `format` as a verb and to mean "CAREFUL: this will change your files!"

As a developer, I would want to be able to run flake8 in one command, which won't alter my files, without running black, which will.

Reply:
I agree that this isn't best. I've raised an issue for us to discuss how to resolve this. (#100) both and the `IBCDFO` level, and at the package level.